### PR TITLE
Demisto Lock & Common Server - support versioned integration context

### PR DESCRIPTION
**Pull request by Itay4**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/37544

## Description
- added demisto version comparison and versioned integration context funcs
- use the versioned integration context in the demisto lock integrations

## Minimum version of Cortex XSOAR
- [x] 6.2.0

## Does it break backward compatibility?
   - [x] No

## Must have
- [x] Tests

### CODE REVIEW

**Review by aviadl — COMMENTED** [review]
You are not passing the version ...

**Comment by Itay4** [comment]
> You are not passing the version ...

@aviadl not sure i follow, it is passed in https://github.com/demisto/content/pull/12960/files#diff-cf15dc446d8fabf3ab46ce1376ace11b9a9ea1a4e435aab4c368513bd48725c0R2166

File: Packs/Base/Scripts/script-CommonServer.yml [hunk 1]
```diff
@@ -2097,7 +2097,77 @@ script: >-
...
+  }
+
+  function isDemistoVersionGE(version, buildNumber) {
+      var serverVersion = {};
+      try {
+          serverVersion = getDemistoVersion();
+          if (serverVersion.version > version) {
+              return true;
+          } else if (serverVersion.version === version) {
+              if (buildNumber !== undefined) {
```
Comment (glicht) [inline Packs/Base/Scripts/script-CommonServer.yml:2115]: I would rather we just check if it not falsy. It usually covers edge cases like someone putting there `null`. 
```suggestion
              if (buildNumber) {
```

File: Packs/Base/Scripts/script-CommonServer.yml [hunk 2]
```diff
@@ -2097,7 +2097,77 @@ script: >-
...
+                  var intBuildNumber = parseInt(serverVersion.buildNumber);
+                  if (isNaN(intBuildNumber)) {
+                      // dev editions are not comparable
+                      logDebug('isDemistoVersionGE: Error.\ninput: server version: {0} build number: {1}\nserver version: {2}'.format(version, buildNumber, JSON.stringify(serverVersion)));
```
Comment (glicht) [inline Packs/Base/Scripts/script-CommonServer.yml:2119]: Better to log everything on one line. 
```suggestion
                      logDebug('isDemistoVersionGE: Error. Input: version: [{0}] buildNumber: [{1}]. server version: {2}'.format(version, buildNumber, JSON.stringify(serverVersion)));
```

File: Packs/Base/TestPlaybooks/TestIsDemistoVersionGE.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+script: |-
+  if (isDemistoVersionGE('1.0.0') === false) {
+      throw new Error('Failed verifying isDemistoVersionGE > 1.0.0');
+  }
+
+  if (isDemistoVersionGE('99.0.0') === true) {
+      throw new Error('Failed verifying isDemistoVersionGE < 99.0.0');
+  }
+
+  if (isDemistoVersionGE('6.2.0', '1') === false) {
```
Comment (glicht) [inline Packs/Base/TestPlaybooks/TestIsDemistoVersionGE.yml:14]: Won't this fail when running the build on server 6.0 or 5.5?

**Comment by glicht** [comment]
@Itay4 need to do some changes here. Need to use the version in getLock and setLock. It might be worth already to put out a from 6.2.0 version as I think the code changes will be significant.

**Comment by Itay4** [comment]
> @Itay4 need to do some changes here. Need to use the version in getLock and setLock. It might be worth already to put out a from 6.2.0 version as I think the code changes will be significant.

done

**Comment by Itay4** [comment]
@glicht i've added the `sync` to be an integration parameter
please review

**Review by glicht — APPROVED** [review]
Looks good. Note that build is failing.

**Pull request merged**